File: src/main/java/org/manapart/microMods/MicroMods.java
package org.manapart.microMods;

import net.minecraftforge.common.MinecraftForge;
import net.minecraftforge.event.entity.player.PlayerInteractEvent;
import net.minecraftforge.eventbus.api.SubscribeEvent;
import net.minecraftforge.fml.common.Mod;
import net.minecraftforge.fml.javafmlmod.FMLJavaModLoadingContext;

@Mod(MicroMods.MODID)
@Mod.EventBusSubscriber(modid = MicroMods.MODID)
public class MicroMods {
    static final String MODID = "micromods";
    private final HorseStats stats = new HorseStats();

    public MicroMods() {
        MinecraftForge.EVENT_BUS.register(this);
    }

    @SubscribeEvent
    public void getStats(PlayerInteractEvent.EntityInteract event) {
        stats.getStats(event);
    }

}

File: src/main/java/org/manapart/microMods/MicroMods.kt
package org.manapart.microMods

import net.minecraft.world.item.Item
import net.minecraftforge.event.entity.player.PlayerInteractEvent
import net.minecraftforge.fml.common.Mod
import thedarkcolour.kotlinforforge.forge.FORGE_BUS
import thedarkcolour.kotlinforforge.forge.MOD_BUS

const val MODID = "micromods"

@Mod(MODID)
object MicroMods {
    private val stats = HorseStats()

    init {
        ModItems.REGISTRY.register(MOD_BUS)
        FORGE_BUS.addListener { event: PlayerInteractEvent.EntityInteract -> stats.getStats(event) }
    }

//    val micromodsIcon = Item(Item.Properties()).also {
//        it.setRegistryName("$MODID:mm_icon")
//    }

}